### PR TITLE
Node id when misbehaving

### PR DESCRIPTION
**Pull request by rebroad**
Displays peer number when a peer is misbehaving. Also, allows logging in situations where a node which has reached it's exceeded allowance of misbehaviour fails to be disconnected.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1611,12 +1614,12 @@ void Misbehaving(NodeId pnode, int howmuch)
 
     state->nMisbehavior += howmuch;
     int banscore = GetArg("-banscore", DEFAULT_BANSCORE_THRESHOLD);
-    if (state->nMisbehavior >= banscore && state->nMisbehavior - howmuch < banscore)
+    if (state->nMisbehavior >= banscore && !state->fShouldBan)
     {
-        LogPrintf("%s: %s (%d -> %d) BAN THRESHOLD EXCEEDED\n", __func__, state->name, state->nMisbehavior-howmuch, state->nMisbehavior);
```
Comment (sipa) [inline src/main.cpp]: No need to add id to state, as CNodeState is exactly indexed by NodeId already (meaning you won't ever have a CNodeState without knowing its NodeId); specifically, pnode here contains the NodeId.

**Comment by laanwj** [comment]
Concept ACK, but indeed adding the id to CNodeState is not necessary.

**Comment by laanwj** [comment]
@rebroad can you please process the comments here so that it can be merged?

**Comment by laanwj** [comment]
Closing this, it seems inactive. The idea of adding the logging is good but it introduces unnecessary state and should be solved differently.

**Pull request closed without merging**